controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -352,6 +352,15 @@ func (r *KubeadmControlPlaneReconciler) upgradeControlPlane(ctx context.Context,
 		return ctrl.Result{}, errors.Wrap(err, "failed to update the kubernetes version in the kubeadm config map")
 	}
 
+	// TODO: Ask vincepri if we should check for a diff in the image fields in
+	// the cluster configuration
+	// TODO: Do something about passing in
+	// kcp.Spec.KubeadmConfigSpec.ClusterConfiguration.DNS. Is there a better
+	// way
+	if err := workloadCluster.UpdateCoreDNSImageInfoInKubeadmConfigMap(ctx, &kcp.Spec.KubeadmConfigSpec.ClusterConfiguration.DNS); err != nil {
```

Comment:
We probably want to move this call to https://github.com/kubernetes-sigs/cluster-api/pull/2574/files#diff-f6190b16a8fe960dc3040118ef7fa34aR253 after the actual upgrade has been performed